### PR TITLE
Add invalidReason to invalid witnesses

### PR DESCRIPTION
**Pull request by danielcolinjames**
This PR adds the new `invalid_reason` property to witnesses, so that it can be displayed on the frontend when using helium-js for anywhere witnesses show up.

I'm new to TypeScript (and Jest) so hopefully I didn't mess anything up; happy to make edits!

### CODE REVIEW

File: packages/http/src/models/Challenge.ts [hunk 1]
```diff
@@ -150,19 +152,23 @@ const constructPath = (path: HTTPPathObject[]): Path[] => {
...
+            packetHash: witness.packet_hash,
+            owner: witness.owner,
+            location: witness.location,
+            isValid: isValidWitness(witness),
+            gateway: witness.gateway,
+            frequency: witness.frequency,
+            datarate: witness.datarate,
+            channel: witness.channel,
+          } as Witness
+          if (!isValidWitness(witness)) witnessObject = { ...witnessObject, invalidReason: witness.invalid_reason }
```
Comment (allenan) [inline packages/http/src/models/Challenge.ts:169]: I think you could do this without defining and mutating an intermediate object:

```js
{
  timestamp: witness.timestamp,
  ...(!isValidWitness(witness)) && { invalidReason: witness.invalid_reason },
}
```

Also, is the `isValidWitness` function call necessary? or could you just check to see if `invalid_reason` is present in the json?

File: packages/http/src/models/__tests__/Challenge.spec.ts [hunk 2]
```diff
@@ -130,6 +134,18 @@ describe('Challenge Model', () => {
...
+  it('has invalid reason when witness is not valid', () => {
+    const challenge = new Challenge(challengeJson([
+      {
+        witnesses: [mockWitness(false)] as HTTPWitnessesObject[],
+        receipt: mockReceipt,
+        geocode: mockGeocode,
+        ...mockPathData,
+      } as HTTPPathObject,
+    ] as HTTPPathObject[]))
+    expect(challenge.path[0].witnesses[0].invalidReason).not.toBeUndefined();
```
Comment (allenan) [inline packages/http/src/models/__tests__/Challenge.spec.ts:146]: might want to test that `invalidReason` equals `fake_invalid_reason`

File: packages/http/src/models/Challenge.ts (hunk 1 above)
Comment (danielcolinjames) [reply packages/http/src/models/Challenge.ts:169]: oh yeah that's way better, I tried to do something like that but couldn't remember the syntax for it, nice! and yeah will just check for witness.invalid_reason instead, good call 👍

**Review by allenan — CHANGES_REQUESTED** [review]
nice work with ts and jest! just a couple small questions/suggestions

File: packages/http/src/models/__tests__/Challenge.spec.ts (hunk 2 above)
Comment (danielcolinjames) [reply packages/http/src/models/__tests__/Challenge.spec.ts:146]: good call!

**Comment by danielcolinjames** [comment]
updated! thanks for the suggestions :)

**Review by allenan — APPROVED** [review]

**Pull request merged**